### PR TITLE
training: fix the learning rate implementation

### PR DESCRIPTION
**Pull request by delhomer**
Before this PR, the learning rate was still implemented following the TensorFlow version of the code. The way the learning rate is considered is now updated: it is introduced as a parameter of the optimizer function, instead of being an attribute of the `ConvolutionalNeuralNetwork` class.

As a direct consequence of this treatment, we do not consider a parsed list of argument any more: a `-L` argument handles the starting learning rate, and a `-l` argument manages the learning rate decay rate.

This PR fixes the issue #60.

### CODE REVIEW

**Review by garaud — APPROVED** [review]
looks good for me
